src/main/scanLogic/scanRunners/applicabilityScan.ts

```diff
@@ -90,20 +91,23 @@ export class ApplicabilityRunner extends BinaryRunner {
         if (issues) {
             // Generate applicable data for all the issues
             issues.forEach(analyzeIssue => {
-                let applicableDetails: CveApplicableDetails = this.getOrCreateApplicableDetails(
-                    analyzeIssue,
-                    applicable,
-                    rulesFullDescription.get(analyzeIssue.ruleId)
-                );
-                analyzeIssue.locations.forEach(location => {
-                    let fileIssues: FileIssues = this.getOrCreateFileIssues(applicableDetails, location.physicalLocation.artifactLocation.uri);
-                    fileIssues.locations.push(location.physicalLocation.region);
-                });
+                if ((!analyzeIssue.kind || analyzeIssue.kind === 'fail') && analyzeIssue.locations) {
```

Comment:
Shorter and safer, since we won't fail if a new unknown state will be introduced in the future
```suggestion
                if (analyzeIssue.kind === 'pass' && analyzeIssue.locations) {
```

Reply:
the SARIF states that if kind attribute is not present it defaults to 'fail'. the 'pass' kind is for CVE that are not applicable, i.e the scan has passed without issues. there are no locations in this case.
It needs to stay as it was to match the SARIF Spec